docs/client.rst

```diff
@@ -0,0 +1,137 @@
+
+Create and enable a new Python3 virtualenv::
+
+    python3 -m venv my_virtualenv
+    source my_virtualenv/bin/activate
+
+Set up the Python dependencies::
+
+    pip install pipenv
+    pipenv install --dev
+    pipenv shell
+
+Install the packaged dependencies::
+
+    sudo apt install python3-pyqt5 python3-pyqt5.qtsvg xvfb
+
+Check you have everything working by running the complete test suite::
+
+    make check
+
+If you see a green report of 100% unit test coverage, you're good to go!
+
+How to Find Help
+----------------
+
+If you would like to report a problem `submit a new issue <https://github.com/freedomofpress/securedrop-client/issues/new>`_.
+
+If you'd like to chat with other developers working on the client drop
+into our `Gitter chat channel for the project <https://gitter.im/freedomofpress/securedrop>`_.
+
+Every week day at 10am (Pacific Time) we take part in a public daily stand-up,
+usually via a `meeting on Jitsi <https://meet.jit.si/QuickWizardsDanceHigh>`_
+(although the details of each daily meeting are published on the Gitter channel
+five minutes before the start of the meeting). All are welcome to contribute.
+
+Otherwise, read on.
+
+Client Architecture
+-------------------
+
+The SecureDrop client is a PyQt application. It's written using Python 3.5 and
+the Python bindings for the Qt UI framework (`PyQt <https://riverbankcomputing.com/software/pyqt/intro>`_).
+
+In the root directory of the repository are two important directories:
+``securedrop_client`` (containing the application code) and ``tests``
+containing our unit tests. You'll also find a Makefile in the root directory
+which defines commands to run commonly needed activities. Type, ``make`` to
+find out what commands are available.
+
+The code in the ``securedrop_client`` namespace is organised in the following
+way:
+
+* ``app.py`` - starts and configures the application.
+* ``logic.py`` - contains the application logic, encapsulated in the ``Client`` class.
+* ``models.py`` - holds all the `SQLAlchemy ORM model definitions <https://www.sqlalchemy.org/>`_ for interacting with the local Sqlite database.
+* ``storage.py`` - contains the functions needed for interacting with a remote SecureDrop API and the local database.
+* ``utils.py`` - generic utility functions needed throughout the application.
+* ``gui`` - this namespace contains two modules: ``main.py`` (containing the ``Window`` class through which all interactions with the user interface should happen) and ``widgets.py`` (containing all the custom widgets used by the ``Window`` class to draw the user interface).
+
+We try very hard to leep the application logic and UI code cleanly separated.
+Furthermore, we try equally hard to ensure the main GUI code always remains
+unblocked. For instance look at how the ``APICallRunner`` is used in
+``logic.py`` to make unblocked network calls to the remote API.
+
+We expect all classes, methods and functions to have docstrings describing the
```

Comment:
We don't enforce this among ourselves and should clean this up before requiring new contributors do do the same. Additionally, we should include `:param xxx:` and `:return:` markup.